Lib/zipfile/_path.py

```diff
@@ -281,9 +281,9 @@ def stem(self):
     def filename(self):
         return pathlib.Path(self.root.filename).joinpath(self.at)
 
-    def read_text(self, *args, **kwargs):
-        kwargs["encoding"] = io.text_encoding(kwargs.get("encoding"))
```

Comment:
I'm beginning to wonder now if this transform is even necessary, given that `self.open` already handles the parameter.

Reply:
Same, it's very strange that it was added to this method at all.  simplified, encoding is handled entirely in open now.